src/worker/startup.ts

```diff
@@ -125,17 +126,25 @@ export async function start() {
 	// exposing queue metrics at a regular interval
 	timer = setInterval(sendQueueMetrics, 60000);
 
-	// Start processing queues
-	queues.discovery.process(5, commonMiddleware(discovery(app, queues), DISCOVERY_LOGGER_NAME));
-	queues.installation.process(
-		Number(CONCURRENT_WORKERS),
-		commonMiddleware(processInstallation(app, queues), INSTALLATION_LOGGER_NAME)
-	);
-	queues.push.process(
-		Number(CONCURRENT_WORKERS),
-		commonMiddleware(processPushJob(app), PUSH_LOGGER_NAME)
-	);
-	queues.metrics.process(1, commonMiddleware(metricsJob, METRICS_LOGGER_NAME));
+	if (firstInitialisation) {
+		logger.info("Initialising queue listeners");
+		// Start processing queues
+		queues.discovery.process(5, commonMiddleware(discovery(app, queues), DISCOVERY_LOGGER_NAME));
+		queues.installation.process(
+			Number(CONCURRENT_WORKERS),
+			commonMiddleware(processInstallation(app, queues), INSTALLATION_LOGGER_NAME)
+		);
+		queues.push.process(
+			Number(CONCURRENT_WORKERS),
+			commonMiddleware(processPushJob(app), PUSH_LOGGER_NAME)
+		);
+		queues.metrics.process(1, commonMiddleware(metricsJob, METRICS_LOGGER_NAME));
+		firstInitialisation = false;
+	} else {
```

Comment:
Instead of this, I would set all queues to paused state when we create them in queues.ts, then add the processor to the queues at the top level of this file, then the start/stop functions just resume/pause the queues.  

Make it simpler to understand and don't have to set state.

Reply:
extra benefit, we can get rid of the `running` var as well.

Reply:
I don't want to play with it too much given that we are going to get rid of Redis soon. It is good enough to stay in the repository for 2 more weeks

Reply:
It's just slightly moving code around to make it simpler and less prone to issues.  It'll be in the repo for more than 2 weeks...